Storage Filesystem as a First Class Feature

abh1sar opened the pull request:
### Description

This PR implements Storage filesystem as a first class feature.
https://cwiki.apache.org/confluence/display/CLOUDSTACK/Storage+Filesystem+as+a+First+Class+Feature

Documentation PR: https://github.com/apache/cloudstack-documentation/pull/420

<!--- Describe your changes in DETAIL - And how has behaviour functionally changed. -->

<!-- For new features, provide link to FS, dev ML discussion etc. -->
<!-- In case of bug fix, the expected and actual behaviours, steps to reproduce. -->

<!-- When "Fixes: #<id>" is specified, the issue/PR will automatically be closed when this PR gets merged -->
<!-- For addressing multiple issues/PRs, use multiple "Fixes: #<id>" -->
<!-- Fixes: # -->

<!--- ******************************************************************************* -->
<!--- NOTE: AUTOMATION USES THE DESCRIPTIONS TO SET LABELS AND PRODUCE DOCUMENTATION. -->
<!--- PLEASE PUT AN 'X' in only **ONE** box -->
<!--- ******************************************************************************* -->

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

### Feature/Enhancement Scale or Bug Severity

#### Feature/Enhancement Scale

- [x] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial


### Screenshots (if appropriate):


### How Has This Been Tested?

<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, and the tests you ran to -->

#### How did you try to break this feature and the system with this change?

<!-- see how your change affects other areas of the code, etc. -->


<!-- Please read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/main/CONTRIBUTING.md) document -->

JoaoJandre commented:
Hey @abhisar

Good to see you actually have a specification for such a big feature. Sadly we have too many PRs that change 2k+ lines with little to none specification. 

I have a few suggestions regarding the feature:

1. On the `file_share` table you have on the spec, I think that we are missing a `created` column, to track when the file_share was created. I would also add a `removed` column, to track when it was removed.
2. On your FSM, add a final state called `Expunged`, that is reached when the `ExpungeOperation` is successful.
3. I know that using shell scripts or python might be tempting, but these tend to be hard to maintain and expand without breaking. I would advise on going back to the idea of doing something similar to what is done for the CPVM/SSVM.

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10514

abh1sar commented:
@blueorangutan test

blueorangutan commented:
@abh1sar a [SL] Trillian-Jenkins test job (ol8 mgmt + kvm-ol8) has been kicked to run smoke tests

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10543

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

abh1sar commented:
@weizhouapache PR is ready for review. Documentation PR is pending which I will link in a few days. Also working on UT and integrations tests.

abh1sar commented:
> Hey @abh1sar
> 
> Good to see you actually have a specification for such a big feature. Sadly we have too many PRs that change 2k+ lines with little to none specification.
> 
> I have a few suggestions regarding the feature:
> 
> 1. On the `file_share` table you have on the spec, I think that we are missing a `created` column, to track when the file_share was created. I would also add a `removed` column, to track when it was removed.
> 2. On your FSM, add a final state called `Expunged`, that is reached when the `ExpungeOperation` is successful.
> 3. I know that using shell scripts or python might be tempting, but these tend to be hard to maintain and expand without breaking. I would advise on going back to the idea of doing something similar to what is done for the CPVM/SSVM.

Hi @JoaoJandre,
I am afraid the specification is not upto date with the design changes that were discussed over the dev ML. Apologies for that.
https://lists.apache.org/thread/rd4sborvzlpsw64o3mq257423ronnq53
I will update the spec asap.

1 and 2. I have the created, removed column and Expunge state in the implementation.
3. We have moved away from VR based model of using shell scripts and python code. Please refer to the ML discussion.
The VM is deployed as a normal user VM and the userdata is used to push some udev rules to the VM.
These udev rules take care of following operations :
- Format and export the FS, first time a data volume is added to the VM
- Resize the FS whenever the volume is resized

We can think about using the agent model in future if required, but using Userdata allows us to implement the basic functionality that is very lightweight in terms of mgmt server - vm communication.
Please let me know your thoughts.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10547

weizhouapache commented on plugins/storage/fileshare/storagefsvm/src/main/java/org/apache/cloudstack/storage/fileshare/lifecycle/StorageFsVmFileShareLifeCycle.java:
```diff
@@ -0,0 +1,329 @@
+        }
+        try {
+            UserVm vm = userVmService.destroyVm(userVM.getId(), true);
+            if (!userVmManager.expunge(userVM)) {
```
the previous line will expunge the vm, right ?

```
UserVm vm = userVmService.destroyVm(userVM.getId(), true);
```

abh1sar replied:
destroyVM doesn't really expunge the VM. expunge() has to be called to expunge it completely. see https://github.com/apache/cloudstack/blob/main/plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterScaleWorker.java#L327

weizhouapache commented on plugins/storage/fileshare/storagefsvm/src/main/java/org/apache/cloudstack/storage/fileshare/lifecycle/StorageFsVmFileShareLifeCycle.java:
```diff
@@ -0,0 +1,329 @@
+        if (serviceOffering.getRamSize() < STORAGEFSVM_MIN_RAM_SIZE.valueIn(zone.getId())) {
+            throw new InvalidParameterValueException("Service offering's ram size should be greater than or equal to " + STORAGEFSVM_MIN_RAM_SIZE.key());
+        }
+        if (serviceOffering.isOfferHA() == false) {
```
```suggestion
        if (!serviceOffering.isOfferHA()) {
```

weizhouapache commented on plugins/storage/fileshare/storagefsvm/src/main/java/org/apache/cloudstack/storage/fileshare/lifecycle/StorageFsVmFileShareLifeCycle.java:
```diff
@@ -0,0 +1,329 @@
+        UserVm vm;
+        String fsVmConfig = getStorageFsVmConfig(fileSystem.toString().toLowerCase());
+        String base64UserData = Base64.encodeBase64String(fsVmConfig.getBytes(com.cloud.utils.StringUtils.getPreferredCharset()));
+        vm = userVmService.createAdvancedVirtualMachine(zone, serviceOffering, template, networkIds, owner, hostName, hostName,
```
maybe check the zone type here ?

(only advanced zone without SG is supported, right ?)

abh1sar replied:
Good idea. Added to validateAndGetZone in FileShareServiceImpl.java

weizhouapache commented on plugins/storage/fileshare/storagefsvm/src/main/resources/conf/fsvm-init.yml:
```diff
@@ -0,0 +1,196 @@
+        echo "$(date '+%Y-%m-%d %H:%M:%S') - $1" >> $LOG_FILE
+      }
+
+      HYPERVISOR_INFO=$(dmesg | grep -i hypervisor)
```
hypervisor can be set in a similar way as `{{ fsvm.filesystem }}`

abh1sar replied:
But how would we know what is the hypervisor type before calling `createAdvancedVirtualMachine` in `createFileShareVM` as host won't be decided yet. If we have multiple clusters with different hypervisors, VM can be deployed on any of them?

shwstppr replied:
@abh1sar based on the code I see, template (and in turn hypervisor) is already selected when `createAdvancedVirtualMachine` is passed so would it make sense to pass same hypervisor type?

weizhouapache commented on server/src/main/java/org/apache/cloudstack/storage/fileshare/FileShareServiceImpl.java:
```diff
@@ -0,0 +1,687 @@
+    @Override
+    public List<Class<?>> getCommands() {
+        final List<Class<?>> cmdList = new ArrayList<>();
+        if (FileShareFeatureEnabled.value() == true) {
```
```suggestion
        if (FileShareFeatureEnabled.value()) {
```

weizhouapache commented on server/src/main/java/org/apache/cloudstack/storage/fileshare/FileShareServiceImpl.java:
```diff
@@ -0,0 +1,687 @@
+            throw new InvalidParameterValueException("Stop file share can be done only if the file share is in " + validStates.toString() + " states");
+        }
+
+        if (cleanup == false) {
```
```suggestion
        if (!cleanup) {
```

weizhouapache commented on server/src/main/java/org/apache/cloudstack/storage/fileshare/FileShareServiceImpl.java:
```diff
@@ -0,0 +1,687 @@
+        fileShareSearchBuilder.and("dataCenterId", fileShareSearchBuilder.entity().getDataCenterId(), SearchCriteria.Op.EQ);
+
+        if (keyword != null) {
+            fileShareSearchBuilder.and().op("keywordName", fileShareSearchBuilder.entity().getName(), SearchCriteria.Op.LIKE);
```
should this be `fileShareSearchBuilder.and().op` or `fileShareSearchBuilder.and(` ?

shwstppr replied:
.op will open a new parenthesis

abh1sar replied:
removed .op and tested

weizhouapache commented on systemvm/debian/opt/cloud/bin/setup/storagefsvm.sh:
```diff
@@ -0,0 +1,61 @@
+
+    rm -f /etc/logrotate.d/cloud
+
+    if [ -f /home/cloud/success ]; then
```
is the file written/updated somewhere ?

maybe write to `/root/success` and use the filename here ?

abh1sar replied:
/home/cloud/success is updated by k8, but it is not being updated for storagefsvm.
how about we use the same filename for storagefsvm as well, and just update it at the end of fsvm-setup?

weizhouapache replied:
yep, that's good

we can use the same file so we do not need to rebuild the systemvm template.

weizhouapache reviewed (commented):
@abh1sar 

overall LGTM
I have not tested yet

left some minor comments

shwstppr commented on api/src/main/java/org/apache/cloudstack/api/response/CapabilitiesResponse.java:
```diff
@@ -124,6 +124,14 @@ public class CapabilitiesResponse extends BaseResponse {
     @Param(description = "the retention time for Instances disks stats", since = "4.18.0")
     private Integer instancesDisksStatsRetentionTime;
 
+    @SerializedName("storagefsvmmincpucount")
```
Can these constants be moved to ApiConstants class?

abh1sar replied:
done

shwstppr commented on api/src/main/java/org/apache/cloudstack/api/command/user/config/ListCapabilitiesCmd.java:
```diff
@@ -69,6 +69,8 @@ public void execute() {
         response.setInstancesStatsUserOnly((Boolean) capabilities.get(ApiConstants.INSTANCES_STATS_USER_ONLY));
         response.setInstancesDisksStatsRetentionEnabled((Boolean) capabilities.get(ApiConstants.INSTANCES_DISKS_STATS_RETENTION_ENABLED));
         response.setInstancesDisksStatsRetentionTime((Integer) capabilities.get(ApiConstants.INSTANCES_DISKS_STATS_RETENTION_TIME));
+        response.setStorageFsVmMinCpuCount((Integer)capabilities.get("storageFsVmMinCpuCount"));
```
Can we use same constants that are used in ApiCmd classes?

abh1sar replied:
yes

shwstppr commented on ui/src/views/storage/CreateFileShare.vue:
```diff
@@ -0,0 +1,492 @@
+        this.providersLoading = false
+      })
+    },
+    fetchCapabilities (id) {
```
we already do listCapabilities call during user login, I guess we can use that
Search for `$store.getters.features.` in UI code

abh1sar replied:
thanks, done

shwstppr commented on ui/src/config/section/compute.js:
```diff
@@ -108,6 +108,7 @@ export default {
           docHelp: 'adminguide/virtual_machines.html#changing-the-vm-name-os-or-group',
           dataView: true,
           popup: true,
+          show: (record) => { return record.vmtype !== 'storagefsvm' },
```
Should we hide the options or should we show some warning in the dialogs? I don't have a strong preference though

abh1sar replied:
These operations are not allowed via API call as well, as they can unpredictably affect file share function. So I think it's better if they are not shown in UI as well.

abh1sar replied:
These options are not allowed via apis calls as well as they can affect file share function unpredictably, so I think it's better to hide them from ui as well.

shwstppr commented on server/src/main/java/org/apache/cloudstack/storage/fileshare/FileShareServiceImpl.java:
```diff
@@ -0,0 +1,687 @@
+    public boolean configure(final String name, final Map<String, Object> params) throws ConfigurationException {
+        Map<String, String> configs = configDao.getConfiguration("management-server", params);
+        String workers = configs.get("expunge.workers");
+        int wrks = NumbersUtil.parseInt(workers, 10);
```
should this be configurable as well?

abh1sar replied:
yes, I can add a new fileshare specific fileshare.expunge.workers setting

shwstppr reviewed (commented):
@abh1sar good work!
I guess the implementation is different from initial design with agents for each share but I feel current way would be more scalable at least in number of fileshares that function in an env.

I've added some initial comments. Will review further.
Will need some unit and automated tests

abh1sar commented on api/src/main/java/com/cloud/storage/VolumeApiService.java:
```diff
@@ -102,8 +102,12 @@ public interface VolumeApiService {
 
     boolean deleteVolume(long volumeId, Account caller);
 
+    Volume changeDiskOfferingForVolumeInternal(Long volumeId, Long newDiskOfferingId, Long newSize, Long newMinIops, Long newMaxIops, boolean autoMigrateVolume, boolean shrinkOk) throws ResourceAllocationException;
```
I think so. FileShareServiceImpl is calling changeDiskOfferingForVolumeInternal via VolumeApiService. Any other way to do that?

abh1sar commented on api/src/main/java/org/apache/cloudstack/api/command/user/offering/ListDiskOfferingsCmd.java:
```diff
@@ -82,6 +82,12 @@ public class ListDiskOfferingsCmd extends BaseListProjectAndAccountResourcesCmd
+            type = CommandType.BOOLEAN,
+            description = "Disk offering is meant for File shares",
+            since = "4.20.0")
+    private Boolean fileShare;
```
I removed all fileShare specific disk offering code, as we discussed that it is not required. This was missed somehow. I have removed it now.

abh1sar commented on api/src/main/java/org/apache/cloudstack/api/command/user/storage/fileshare/CreateFileShareCmd.java:
```diff
@@ -0,0 +1,317 @@
+            description = "the disk offering to use for the underlying storage.")
+    private Long diskOfferingId;
+
+    @Parameter(name = ApiConstants.MIN_IOPS,
```
yes, if the diskoffering has customizedIops, then createFIleShareCmd has to take these two values as well.

abh1sar commented on api/src/main/java/org/apache/cloudstack/storage/fileshare/FileShareLifeCycle.java:
```diff
@@ -0,0 +1,40 @@
+public interface FileShareLifeCycle {
+    void checkPrerequisites(DataCenter zone, Long serviceOfferingId);
+
+    Pair<Long, Long> commitFileShare(FileShare fileShare, Long networkId, Long diskOfferingId, Long size, Long minIops, Long maxIops) throws ResourceUnavailableException, InsufficientCapacityException, ResourceAllocationException;
```
fileShareServieImpl calls these methods via the interface. That's why I have put these here.

abh1sar commented on systemvm/debian/opt/cloud/bin/setup/storagefsvm.sh:
```diff
@@ -0,0 +1,61 @@
+    echo "export PATH='$PATH:/opt/bin/'">> ~/.bashrc
+
+    disable_rpfilter
+    enable_fwding 1
```
csknode sets this to 1, as well as virtual router.
I guess IP forwarding should be enabled for storagefsvm as well as it can have nics from multiple networks?

weizhouapache replied:
cksnode and vr act as the gateway , they need to forward the packets from containers and vms
the storagefs vm does not have such requirements (similar as cpvm and ssvm).

abh1sar replied:
sure, will remove this

abh1sar commented on api/src/main/java/org/apache/cloudstack/api/command/user/storage/fileshare/CreateFileShareCmd.java:
```diff
@@ -0,0 +1,168 @@
+
+    @Parameter(name = ApiConstants.FORMAT,
+            type = CommandType.STRING,
+            description = "the filesystem format which will be installed on the file share.")
```
Added supported formats (ext4, xfs)

abh1sar commented on api/src/main/java/org/apache/cloudstack/api/command/user/storage/fileshare/CreateFileShareCmd.java:
```diff
@@ -0,0 +1,168 @@
+
+    @Parameter(name = ApiConstants.PROVIDER,
+            type = CommandType.STRING,
+            description = "the provider to be used for the file share.")
```
done

abh1sar commented on api/src/main/java/org/apache/cloudstack/api/command/user/storage/fileshare/CreateFileShareCmd.java:
```diff
@@ -0,0 +1,168 @@
+            collectionType = CommandType.UUID,
+            entityType = NetworkResponse.class,
+            description = "list of network ids to attach file share to")
+    private List<Long> networkIds;
```
no, just networkid

abh1sar commented on api/src/main/java/org/apache/cloudstack/storage/fileshare/FileShareService.java:
```diff
@@ -0,0 +1,41 @@
+
+public interface FileShareService extends PluggableService {
+
+    boolean configure(String name, Map<String, Object> params) throws ConfigurationException;
```
yes, rewmoved

abh1sar commented on api/src/main/java/org/apache/cloudstack/storage/fileshare/FileShareService.java:
```diff
@@ -0,0 +1,41 @@
+
+    boolean configure(String name, Map<String, Object> params) throws ConfigurationException;
+
+    List<Class<?>> getCommands();
```
copy

abh1sar commented on api/src/main/java/org/apache/cloudstack/api/command/user/storage/fileshare/ChangeFileShareDiskOfferingCmd.java:
```diff
@@ -0,0 +1,149 @@
+    FileShareService fileShareService;
+
+    @Inject
+    protected AccountService accountService;
```
yes, done

abh1sar commented on api/src/main/java/org/apache/cloudstack/api/command/user/storage/fileshare/ChangeFileShareDiskOfferingCmd.java:
```diff
@@ -0,0 +1,149 @@
+    FileShareService fileShareService;
+
+    @Inject
+    protected AccountService accountService;
```
removed this

abh1sar commented on api/src/main/java/org/apache/cloudstack/api/command/user/storage/fileshare/ChangeFileShareServiceOfferingCmd.java:
```diff
@@ -0,0 +1,151 @@
+
+    @Override
+    public String getEventType() {
+        return EventTypes.EVENT_FILESHARE_UPDATE;
```
done

abh1sar commented on api/src/main/java/org/apache/cloudstack/api/response/FileShareResponse.java:
```diff
@@ -0,0 +1,360 @@
+    private String diskOfferingDisplayText;
+
+    @SerializedName(ApiConstants.SERVICE_OFFERING_ID)
+    @Param(description = "service offering for the file share")
```
done

abh1sar commented on api/src/main/java/org/apache/cloudstack/api/response/FileShareResponse.java:
```diff
@@ -0,0 +1,360 @@
+    private String sizeGB;
+
+    @SerializedName(ApiConstants.DISK_OFFERING_ID)
+    @Param(description = "disk offering for the file share")
```
done

shwstppr commented on ui/src/views/storage/FileShareTab.vue:
```diff
@@ -0,0 +1,202 @@
+            </template>
+            <timeline>
+              <a-timeline-item color="blue">
+                <h3>mount -t nfs -o rw,sync {{ nic.ipaddress }}:{{ resource.path }} [local_mount_path]</h3>
```
would be great to get a copy button for commands but it is just UX and can be added later

abh1sar replied:
I tried but couldn't find out how to do it. Will take your help on this.

shwstppr commented on ui/src/components/view/ListView.vue:
```diff
@@ -675,15 +675,15 @@ export default {
         '/zone', '/pod', '/cluster', '/host', '/storagepool', '/imagestore', '/systemvm', '/router', '/ilbvm', '/annotation',
         '/computeoffering', '/systemoffering', '/diskoffering', '/backupoffering', '/networkoffering', '/vpcoffering',
-        '/tungstenfabric', '/oauthsetting', '/guestos', '/guestoshypervisormapping', '/webhook', 'webhookdeliveries'].join('|'))
+        '/tungstenfabric', '/oauthsetting', '/guestos', '/guestoshypervisormapping', '/webhook', 'webhookdeliveries', 'fileshare'].join('|'))
```
```suggestion
        '/tungstenfabric', '/oauthsetting', '/guestos', '/guestoshypervisormapping', '/webhook', '/webhookdeliveries', '/fileshare'].join('|'))
```
for consistency sake, I started this divergence :man_facepalming:

shwstppr commented on server/src/main/java/com/cloud/server/ManagementServerImpl.java:
```diff
@@ -4470,6 +4471,9 @@ public Map<String, Object> listCapabilities(final ListCapabilitiesCmd cmd) {
             regionSecondaryEnabled = true;
         }
 
+        final Integer fsVmMinCpu = Integer.parseInt(_configDao.getValue("storagefsvm.min.cpu.count"));
+        final Integer fsVmMinRam = Integer.parseInt(_configDao.getValue("storagefsvm.min.ram.size"));
```
Not possible to use configkeys?

abh1sar replied:
StorageFsVmFileShareProvider class in plugins is not directly accessible from here. We can put the settings in FileShareProvider but that won't be totally accurate as these settings are only pertaining to storagefsvm provider.

shwstppr commented on server/src/main/java/com/cloud/storage/snapshot/SnapshotManagerImpl.java:
```diff
@@ -983,7 +983,7 @@ public SnapshotPolicyVO createPolicy(CreateSnapshotPolicyCmd cmd, Account policy
                 userVmVO = _vmDao.findById(instanceId);
             }
-            if (template != null && template.getTemplateType() == Storage.TemplateType.SYSTEM && (userVmVO == null || !UserVmManager.CKS_NODE.equals(userVmVO.getUserVmType()))) {
+            if (template != null && template.getTemplateType() == Storage.TemplateType.SYSTEM && (userVmVO == null || !UserVmManager.CKS_NODE.equals(userVmVO.getUserVmType()) || !UserVmManager.STORAGEFSVM.equals(userVmVO.getUserVmType()))) {
```
Not within scope of this PR but there should be some common framework to check operation support for different uservm types

abh1sar replied:
Yes, that's a good idea

shwstppr reviewed (commented):
@abh1sar some comments but looks good to me

abh1sar commented:
@blueorangutan package

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10598

abh1sar commented:
@blueorangutan test

blueorangutan commented:
@abh1sar a [SL] Trillian-Jenkins test job (ol8 mgmt + kvm-ol8) has been kicked to run smoke tests

shwstppr commented on plugins/storage/fileshare/storagefsvm/src/main/java/org/apache/cloudstack/storage/fileshare/lifecycle/StorageFsVmFileShareLifeCycle.java:
```diff
@@ -0,0 +1,313 @@
+        ServiceOffering serviceOffering = serviceOfferingDao.findById(serviceOfferingId);
+
+        DataCenter zone = dataCenterDao.findById(zoneId);
+        Hypervisor.HypervisorType availableHypervisor = resourceMgr.getAvailableHypervisor(zoneId);
```
@abh1sar can you please check behaviour with a multi-hypervisor env where the default hypervisor gets out of capacity?

I feel we should
- get the list of all available hypervisors and iterate over them
- check if the required template is present
- check if a deployment plan is possible

abh1sar replied:
Done.

blueorangutan commented:
<b>[SF] Trillian test result (tid-11047)</b>
Environment: kvm-ol8 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 54625 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11047-kvm-ol8.zip
Smoke tests completed. 138 look OK, 1 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_redundant_vpc_site2site_vpn | `Failure` | 439.46 | test_vpc_vpn.py

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10611

shwstppr commented on plugins/storage/fileshare/storagefsvm/src/main/java/org/apache/cloudstack/storage/fileshare/lifecycle/StorageFsVmFileShareLifeCycle.java:
```diff
@@ -171,15 +171,56 @@ private UserVm createFileShareVM(Long zoneId, Account owner, List<Long> networkI
+                        null, null, keypairs, null, addrs, null, null, null,
+                        customParameterMap, null, null, null, null,
+                        true, UserVmManager.STORAGEFSVM, null);
+            } catch (InsufficientCapacityException ex) {
+                if (iter.hasNext()) {
+                    continue;
+                } else {
+                    throw ex;
+                }
+            } finally {
+                CallContext.unregister();
+            }
+
+            vmContext = CallContext.register(CallContext.current(), ApiCommandResourceType.VirtualMachine);
+            vmContext.setEventResourceId(vm.getId());
+            try {
+                userVmService.startVirtualMachine(vm);
+            } catch (InsufficientCapacityException ex) {
+                expungeVm(vm.getId());
+                if (iter.hasNext()) {
+                    continue;
+                } else {
+                    throw ex;
+                }
+            } finally {
+                CallContext.unregister();
+            }
```
would it make sense to merge these blocks? Something like,

```
                vmContext.setEventResourceId(vm.getId());
                userVmService.startVirtualMachine(vm);
            } catch (InsufficientCapacityException ex) {
                if (vm != null) {
                    expungeVm(vm.getId());
                }
                if (iter.hasNext()) {
                    continue;
                } else {
                    throw ex;
                }
            } finally {
                CallContext.unregister();
            }
```

abh1sar replied:
yes, this looks much better.

shwstppr reviewed (approved):
code looks good. One minor comment regarding code structure

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10633

abh1sar commented:
@blueorangutan test

blueorangutan commented:
@abh1sar a [SL] Trillian-Jenkins test job (ol8 mgmt + kvm-ol8) has been kicked to run smoke tests

blueorangutan commented:
<b>[SF] Trillian test result (tid-11069)</b>
Environment: kvm-ol8 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 52201 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11069-kvm-ol8.zip
Smoke tests completed. 139 look OK, 1 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_mount_file_share | `Failure` | 391.80 | test_file_share_lifecycle.py

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✖️ el8 ✖️ el9 ✖️ debian ✖️ suse15. SL-JID 10648

abh1sar commented:
@blueorangutan package

blueorangutan commented:
Packaging result [SF]: ✖️ el8 ✖️ el9 ✖️ debian ✖️ suse15. SL-JID 10653

abh1sar commented:
@blueorangutan package

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10655

borisstoyanov commented:
@abh1sar, could you please check the merge conflicts when you have time

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10723

DaanHoogland commented on api/src/main/java/com/cloud/event/EventTypes.java:
```diff
@@ -745,7 +745,7 @@ public class EventTypes {
     public static final String EVENT_FILESHARE_START = "FILESHARE.START";
     public static final String EVENT_FILESHARE_UPDATE = "FILESHARE.UPDATE";
-    public static final String EVENT_FILESHARE_CHANGE_SVC_OFFERING = "FILESHARE.CHANGE.SERVICE.OFFERING";
+    public static final String EVENT_FILESHARE_CHANGE_SVC_OFFERING = "FILESHARE.CHANGE.SVC.OFFERING";
```
why make this name more obscure?

abh1sar replied:
The string length was getting more than that allowed in the db table. But have renamed it now to SHAREDFS.CHANGE.SERVICE.OFFERING

borisstoyanov commented:
@blueorangutan test matrix

blueorangutan commented:
<b>[SF] Trillian test result (tid-11128)</b>
Environment: kvm-ol8 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 47928 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11128-kvm-ol8.zip
Smoke tests completed. 137 look OK, 3 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_DeployVmAffinityGroup | `Error` | 1.42 | test_affinity_groups.py
ContextSuite context=TestFileShareLifecycle>:setup | `Error` | 0.00 | test_file_share_lifecycle.py
test_01_non_strict_host_anti_affinity | `Error` | 3.58 | test_nonstrict_affinity_group.py
test_02_non_strict_host_affinity | `Error` | 3.62 | test_nonstrict_affinity_group.py

blueorangutan commented:
<b>[SF] Trillian test result (tid-11129)</b>
Environment: kvm-ubuntu22 (x2), Advanced Networking with Mgmt server u22
Total time taken: 56776 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11129-kvm-ubuntu22.zip
Smoke tests completed. 136 look OK, 4 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_DeployVmAffinityGroup | `Error` | 1.43 | test_affinity_groups.py
ContextSuite context=TestClusterDRS>:setup | `Error` | 0.00 | test_cluster_drs.py
ContextSuite context=TestFileShareLifecycle>:setup | `Error` | 0.00 | test_file_share_lifecycle.py
test_01_non_strict_host_anti_affinity | `Error` | 3.62 | test_nonstrict_affinity_group.py
test_02_non_strict_host_affinity | `Error` | 3.63 | test_nonstrict_affinity_group.py

blueorangutan commented:
<b>[SF] Trillian test result (tid-11131)</b>
Environment: xcpng82 (x2), Advanced Networking with Mgmt server a9
Total time taken: 65454 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11131-xcpng82.zip
Smoke tests completed. 135 look OK, 5 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_DeployVmAffinityGroup | `Error` | 1.36 | test_affinity_groups.py
test_01_condensed_drs_algorithm | `Failure` | 164.46 | test_cluster_drs.py
test_02_balanced_drs_algorithm | `Failure` | 187.56 | test_cluster_drs.py
ContextSuite context=TestFileShareLifecycle>:setup | `Error` | 0.00 | test_file_share_lifecycle.py
test_01_non_strict_host_anti_affinity | `Error` | 2.57 | test_nonstrict_affinity_group.py
test_02_non_strict_host_affinity | `Error` | 3.57 | test_nonstrict_affinity_group.py
test_02_create_volume | `Error` | 4.30 | test_resource_names.py

borisstoyanov commented:
@abh1sar it seems like the test_file_share_lifecycle is failing consistently across all hypervisors, maybe something wrong with the test itself. 
`CmdName: createFileShare Parameter : filesystem is Required CSLog: ERROR: marvinRequest : CmdName: <marvin.cloudstackAPI.createFileShare.createFileShareCmd object at 0x7fafa2ab4b38> Exception: ['Traceback (most recent call last):\n', ' File "/usr/local/lib/python3.6/site-packages/marvin/cloudstackConnection.py", line 355, in marvinRequest\n raise self.__lastError\n', 'marvin.cloudstackException.InvalidParameterException: Invalid Parameters\n'] Traceback (most recent call last): File "/usr/local/lib/python3.6/site-packages/marvin/cloudstackConnection.py", line 355, in marvinRequest raise self.__lastError marvin.cloudstackException.InvalidParameterException: Invalid Parameters CSLog: `

blueorangutan commented:
<b>[SF] Trillian test result (tid-11130)</b>
Environment: vmware-70u3 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 104691 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11130-vmware-70u3.zip
Smoke tests completed. 132 look OK, 8 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_DeployVmAffinityGroup | `Error` | 1.38 | test_affinity_groups.py
ContextSuite context=TestFileShareLifecycle>:setup | `Error` | 0.00 | test_file_share_lifecycle.py
test_list_vms_metrics_admin | `Error` | 3625.06 | test_metrics_api.py
test_list_vms_metrics_history | `Error` | 5.80 | test_metrics_api.py
test_list_volumes_metrics_history | `Error` | 3627.64 | test_metrics_api.py
test_01_non_strict_host_anti_affinity | `Error` | 3.60 | test_nonstrict_affinity_group.py
test_02_non_strict_host_affinity | `Error` | 3.61 | test_nonstrict_affinity_group.py
test_02_restore_vm_with_disk_offering | `Error` | 65.57 | test_restore_vm.py
test_03_restore_vm_with_disk_offering_custom_size | `Error` | 58.28 | test_restore_vm.py
test_01_sys_vm_start | `Failure` | 0.10 | test_secondary_storage.py
test_03_live_migrate_VM_with_two_data_disks | `Error` | 63.30 | test_vm_life_cycle.py
test_08_migrate_vm | `Error` | 51.32 | test_vm_life_cycle.py
test_01_migrate_vm_strict_tags_success | `Error` | 3603.18 | test_vm_strict_host_tags.py
test_02_migrate_vm_strict_tags_failure | `Error` | 3.82 | test_vm_strict_host_tags.py
test_01_restore_vm_strict_tags_success | `Error` | 19.19 | test_vm_strict_host_tags.py
test_02_restore_vm_strict_tags_failure | `Error` | 3608.00 | test_vm_strict_host_tags.py
test_01_scale_vm_strict_tags_success | `Error` | 17.23 | test_vm_strict_host_tags.py
test_02_scale_vm_strict_tags_failure | `Error` | 3604.72 | test_vm_strict_host_tags.py
test_01_deploy_vm_on_specific_host_without_strict_tags | `Error` | 20.18 | test_vm_strict_host_tags.py
test_02_deploy_vm_on_any_host_without_strict_tags | `Error` | 3606.61 | test_vm_strict_host_tags.py
test_03_deploy_vm_on_specific_host_with_strict_tags_success | `Error` | 9.14 | test_vm_strict_host_tags.py
test_04_deploy_vm_on_any_host_with_strict_tags_success | `Error` | 3659.91 | test_vm_strict_host_tags.py

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10792

DaanHoogland commented:
@blueorangutan test

blueorangutan commented:
<b>[SF] Trillian Build Failed (tid-11182)<b/>

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10833

abh1sar commented:
@blueorangutan test

blueorangutan commented:
@abh1sar a [SL] Trillian-Jenkins test job (ol8 mgmt + kvm-ol8) has been kicked to run smoke tests

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10855

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10864

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10871

blueorangutan commented:
<b>[SF] Trillian test result (tid-11236)</b>
Environment: kvm-ol8 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 55148 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11236-kvm-ol8.zip
Smoke tests completed. 138 look OK, 2 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_03_network_operations_on_created_vm_of_otheruser | `Failure` | 98.16 | test_network_permissions.py
test_03_network_operations_on_created_vm_of_otheruser | `Error` | 98.17 | test_network_permissions.py
test_01_nic | `Error` | 176.26 | test_nic.py

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10889

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10890

blueorangutan commented:
<b>[SF] Trillian Build Failed (tid-11267)<b/>

blueorangutan commented:
<b>[SF] Trillian Build Failed (tid-11265)<b/>

blueorangutan commented:
<b>[SF] Trillian Build Failed (tid-11269)<b/>

blueorangutan commented:
<b>[SF] Trillian test result (tid-11266)</b>
Environment: kvm-ol8 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 52757 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11266-kvm-ol8.zip
Smoke tests completed. 139 look OK, 1 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_05_rvpc_multi_tiers | `Failure` | 526.66 | test_vpc_redundant.py
test_05_rvpc_multi_tiers | `Error` | 526.68 | test_vpc_redundant.py

blueorangutan commented:
<b>[SF] Trillian test result (tid-11270)</b>
Environment: vmware-70u3 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 62583 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11270-vmware-70u3.zip
Smoke tests completed. 137 look OK, 3 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_deployVMInSharedNetwork | `Failure` | 3606.07 | test_network.py
ContextSuite context=TestSharedNetworkWithConfigDrive>:teardown | `Error` | 3607.30 | test_network.py
test_02_restore_vm_with_disk_offering | `Error` | 65.35 | test_restore_vm.py
test_03_restore_vm_with_disk_offering_custom_size | `Error` | 58.18 | test_restore_vm.py
test_resize_shared_fs | `Failure` | 32.51 | test_sharedfs_lifecycle.py

blueorangutan commented:
<b>[SF] Trillian test result (tid-11268)</b>
Environment: xcpng82 (x2), Advanced Networking with Mgmt server a9
Total time taken: 71930 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11268-xcpng82.zip
Smoke tests completed. 135 look OK, 5 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
ContextSuite context=TestClusterDRS>:setup | `Error` | 0.00 | test_cluster_drs.py
ContextSuite context=TestSharedNetworkWithConfigDrive>:setup | `Error` | 7.48 | test_network.py
test_01_non_strict_host_anti_affinity | `Error` | 219.22 | test_nonstrict_affinity_group.py
test_02_non_strict_host_affinity | `Error` | 118.38 | test_nonstrict_affinity_group.py
test_02_create_volume | `Error` | 2.25 | test_resource_names.py
test_mount_shared_fs | `Failure` | 300.40 | test_sharedfs_lifecycle.py
test_resize_shared_fs | `Failure` | 300.39 | test_sharedfs_lifecycle.py

weizhouapache commented:
@blueorangutan package

blueorangutan commented:
@weizhouapache a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

weizhouapache commented on engine/schema/src/main/resources/META-INF/db/schema-41910to42000.sql:
```diff
@@ -158,6 +158,298 @@ WHERE
+    INDEX `i_shared_filesystem__domain_id`(`domain_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+DROP VIEW IF EXISTS `cloud`.`shared_filesystem_view`;
```
these views should be removed from schema-41910to42000.sql

abh1sar replied:
Done

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10896

abh1sar commented:
@blueorangutan package

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10897

blueorangutan commented:
<b>[SF] Trillian test result (tid-11277)</b>
Environment: vmware-70u3 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 73433 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11277-vmware-70u3.zip
Smoke tests completed. 135 look OK, 5 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_list_vms_metrics_admin | `Error` | 3623.74 | test_metrics_api.py
test_list_vms_metrics_history | `Error` | 5.67 | test_metrics_api.py
test_list_volumes_metrics_history | `Error` | 3617.54 | test_metrics_api.py
test_01_deployVMInSharedNetwork | `Failure` | 3605.35 | test_network.py
ContextSuite context=TestSharedNetworkWithConfigDrive>:teardown | `Error` | 3606.58 | test_network.py
test_02_restore_vm_with_disk_offering | `Error` | 56.23 | test_restore_vm.py
test_03_restore_vm_with_disk_offering_custom_size | `Error` | 56.24 | test_restore_vm.py
test_02_isolate_network_FW_PF_default_routes_egress_false | `Failure` | 145.04 | test_routers_network_ops.py
test_resize_shared_fs | `Failure` | 32.52 | test_sharedfs_lifecycle.py

blueorangutan commented:
<b>[SF] Trillian Build Failed (tid-11298)<b/>

blueorangutan commented:
<b>[SF] Trillian Build Failed (tid-11302)<b/>

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✖️ debian ✔️ suse15. SL-JID 10917

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10919

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10920

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

weizhouapache commented on systemvm/debian/opt/cloud/bin/setup/cloud-early-config:
```diff
@@ -63,7 +63,7 @@ patch() {
   retry=60
   local patched=false
-  if [ "$TYPE" != "cksnode" ]; then
+  if [ "$TYPE" == "cksnode" ] || [ "$TYPE" == "sharedfsvm" ]; then
```
```suggestion
  if [ "$TYPE" != "cksnode" ] && [ "$TYPE" != "sharedfsvm" ]; then
```

abh1sar replied:
Done

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10923

weizhouapache commented on systemvm/debian/opt/cloud/bin/setup/sharedfsvm.sh:
```diff
@@ -0,0 +1,69 @@
+. /opt/cloud/bin/setup/common.sh
+
+setup_sharedfsvm() {
+    log_it "Starting cloud-init services"
```
```suggestion
```

same as line 56

abh1sar replied:
Done

weizhouapache commented on ui/src/config/section/storage.js:
```diff
@@ -543,6 +543,142 @@ export default {
+          message: 'message.action.stop.sharedfs',
+          docHelp: 'adminguide/storage.html#lifecycle-operations',
+          dataView: true,
+          popup: true,
```
```suggestion
          popup: true,
          groupAction: true,
```

abh1sar replied:
Done

borisstoyanov commented:
@blueorangutan package

blueorangutan commented:
@borisstoyanov a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10932

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

weizhouapache commented on systemvm/debian/opt/cloud/bin/setup/sharedfsvm.sh:
```diff
@@ -0,0 +1,68 @@
+    sed -i '/- ssh$/s/- ssh/- [ssh, always]/' /etc/cloud/cloud.cfg
+
+    # Prevent root login
+    > /root/.ssh/authorized_keys
```
```suggestion
    > /root/.ssh/authorized_keys
    > /home/debian/.ssh/authorized_keys

    rm -f /etc/cloud/cloud-init.disabled

```

weizhouapache replied:
as discussed, let's do it later to support ssh keys

abh1sar commented:
@blueorangutan package

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10936

blueorangutan commented:
<b>[SF] Trillian test result (tid-11307)</b>
Environment: vmware-70u3 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 61425 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11307-vmware-70u3.zip
Smoke tests completed. 137 look OK, 3 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_deploy_more_vms_than_limit_allows | `Error` | 140.24 | test_deploy_vms_in_parallel.py
test_01_deployVMInSharedNetwork | `Failure` | 3602.99 | test_network.py
ContextSuite context=TestSharedNetworkWithConfigDrive>:teardown | `Error` | 3604.27 | test_network.py
test_02_restore_vm_with_disk_offering | `Error` | 66.45 | test_restore_vm.py
test_03_restore_vm_with_disk_offering_custom_size | `Error` | 56.26 | test_restore_vm.py

blueorangutan commented:
@borisstoyanov a [SL] Trillian-Jenkins matrix job (EL8 mgmt + EL8 KVM, Ubuntu22 mgmt + Ubuntu22 KVM, EL8 mgmt + VMware 7.0u3, EL9 mgmt + XCP-ng 8.2 ) has been kicked to run smoke tests

weizhouapache reviewed (approved):
code lgtm

@abh1sar 
can you fixn the conflicts ?

rohityadavcloud commented:
@blueorangutan package

blueorangutan commented:
@rohityadavcloud a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10954

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
<b>[SF] Trillian test result (tid-11315)</b>
Environment: xcpng82 (x2), Advanced Networking with Mgmt server a9
Total time taken: 69612 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11315-xcpng82.zip
Smoke tests completed. 131 look OK, 9 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_condensed_drs_algorithm | `Failure` | 182.81 | test_cluster_drs.py
test_02_balanced_drs_algorithm | `Failure` | 176.82 | test_cluster_drs.py
ContextSuite context=TestSharedNetworkWithConfigDrive>:setup | `Error` | 15.76 | test_network.py
test_01_non_strict_host_anti_affinity | `Error` | 218.41 | test_nonstrict_affinity_group.py
test_02_non_strict_host_affinity | `Error` | 116.44 | test_nonstrict_affinity_group.py
test_02_create_volume | `Error` | 4.28 | test_resource_names.py
test_01_sys_vm_start | `Failure` | 0.10 | test_secondary_storage.py
ContextSuite context=TestSharedFSLifecycle>:setup | `Error` | 0.00 | test_sharedfs_lifecycle.py
test_01_deploy_vm_on_specific_host | `Error` | 0.10 | test_vm_deployment_planner.py
test_04_deploy_vm_on_host_override_pod_and_cluster | `Error` | 0.13 | test_vm_deployment_planner.py
test_08_migrate_vm | `Error` | 0.08 | test_vm_life_cycle.py
test_01_migrate_vm_strict_tags_success | `Error` | 0.35 | test_vm_strict_host_tags.py
test_02_migrate_vm_strict_tags_failure | `Error` | 0.37 | test_vm_strict_host_tags.py
test_01_restore_vm_strict_tags_success | `Error` | 0.31 | test_vm_strict_host_tags.py
test_02_restore_vm_strict_tags_failure | `Error` | 0.31 | test_vm_strict_host_tags.py
test_01_scale_vm_strict_tags_success | `Error` | 0.28 | test_vm_strict_host_tags.py
test_02_scale_vm_strict_tags_failure | `Error` | 0.32 | test_vm_strict_host_tags.py
test_01_deploy_vm_on_specific_host_without_strict_tags | `Error` | 0.30 | test_vm_strict_host_tags.py
test_02_deploy_vm_on_any_host_without_strict_tags | `Error` | 5.76 | test_vm_strict_host_tags.py
test_03_deploy_vm_on_specific_host_with_strict_tags_success | `Error` | 0.31 | test_vm_strict_host_tags.py
test_04_deploy_vm_on_any_host_with_strict_tags_success | `Error` | 2.79 | test_vm_strict_host_tags.py
test_05_deploy_vm_on_specific_host_with_strict_tags_failure | `Failure` | 0.39 | test_vm_strict_host_tags.py

blueorangutan commented:
<b>[SF] Trillian test result (tid-11333)</b>
Environment: xcpng82 (x2), Advanced Networking with Mgmt server a9
Total time taken: 51016 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11333-xcpng82.zip
Smoke tests completed. 115 look OK, 25 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
ContextSuite context=TestClusterDRS>:setup | `Error` | 0.00 | test_cluster_drs.py
test_01_internallb_roundrobin_1VPC_3VM_HTTP_port80 | `Failure` | 3.92 | test_internal_lb.py
test_02_internallb_roundrobin_1RVPC_3VM_HTTP_port80 | `Failure` | 3.83 | test_internal_lb.py
test_03_vpc_internallb_haproxy_stats_on_all_interfaces | `Failure` | 4.91 | test_internal_lb.py
test_04_rvpc_internallb_haproxy_stats_on_all_interfaces | `Failure` | 3.77 | test_internal_lb.py
test_01_invalid_upgrade_kubernetes_cluster | `Failure` | 2.36 | test_kubernetes_clusters.py
test_02_upgrade_kubernetes_cluster | `Failure` | 2.37 | test_kubernetes_clusters.py
test_03_deploy_and_scale_kubernetes_cluster | `Failure` | 2.39 | test_kubernetes_clusters.py
test_04_autoscale_kubernetes_cluster | `Failure` | 2.37 | test_kubernetes_clusters.py
test_05_basic_lifecycle_kubernetes_cluster | `Failure` | 2.35 | test_kubernetes_clusters.py
test_06_delete_kubernetes_cluster | `Failure` | 2.37 | test_kubernetes_clusters.py
test_08_upgrade_kubernetes_ha_cluster | `Failure` | 2.49 | test_kubernetes_clusters.py
test_10_vpc_tier_kubernetes_cluster | `Error` | 3.68 | test_kubernetes_clusters.py
test_11_test_unmanaged_cluster_lifecycle | `Error` | 4.79 | test_kubernetes_clusters.py
ContextSuite context=TestListIdsParams>:setup | `Error` | 0.00 | test_list_ids_parameter.py
ContextSuite context=TestListVolumes>:setup | `Error` | 0.00 | test_list_volumes.py
test_09_project_suspend | `Error` | 2.58 | test_projects.py
test_10_project_activation | `Error` | 2.42 | test_projects.py
ContextSuite context=TestLoadBalance>:setup | `Error` | 0.00 | test_loadbalance.py
test_list_system_vms_metrics_history | `Failure` | 0.26 | test_metrics_api.py
test_list_vms_metrics_admin | `Error` | 2.08 | test_metrics_api.py
test_list_vms_metrics_history | `Error` | 1.73 | test_metrics_api.py
test_list_vms_metrics_user | `Error` | 1.77 | test_metrics_api.py
test_list_volumes_metrics_history | `Error` | 1.61 | test_metrics_api.py
test_nic_secondaryip_add_remove | `Error` | 21.08 | test_multipleips_per_nic.py
test_network_acl | `Error` | 2.43 | test_network_acl.py
test_01_verify_ipv6_network | `Error` | 3.33 | test_network_ipv6.py
test_01_verify_ipv6_network | `Error` | 3.33 | test_network_ipv6.py
test_03_network_operations_on_created_vm_of_otheruser | `Error` | 2.66 | test_network_permissions.py
test_03_network_operations_on_created_vm_of_otheruser | `Error` | 2.66 | test_network_permissions.py
test_04_deploy_vm_for_other_user_and_test_vm_operations | `Failure` | 1.49 | test_network_permissions.py
ContextSuite context=TestNetworkPermissions>:teardown | `Error` | 1.59 | test_network_permissions.py
test_delete_account | `Error` | 20.47 | test_network.py
test_delete_network_while_vm_on_it | `Error` | 2.58 | test_network.py
test_delete_network_while_vm_on_it | `Error` | 2.58 | test_network.py
test_deploy_vm_l2network | `Error` | 2.54 | test_network.py
test_deploy_vm_l2network | `Error` | 2.54 | test_network.py
test_l2network_restart | `Error` | 3.79 | test_network.py
test_l2network_restart | `Error` | 3.79 | test_network.py
ContextSuite context=TestL2Networks>:teardown | `Error` | 4.92 | test_network.py
ContextSuite context=TestPortForwarding>:setup | `Error` | 9.77 | test_network.py
ContextSuite context=TestPublicIP>:setup | `Error` | 8.13 | test_network.py
test_reboot_router | `Error` | 5.07 | test_network.py
test_releaseIP | `Error` | 4.57 | test_network.py
test_releaseIP_using_IP | `Error` | 4.91 | test_network.py
ContextSuite context=TestRouterRules>:setup | `Error` | 10.48 | test_network.py
test_01_deployVMInSharedNetwork | `Failure` | 1.28 | test_network.py
test_02_verifyRouterIpAfterNetworkRestart | `Failure` | 1.11 | test_network.py
test_03_destroySharedNetwork | `Failure` | 1.10 | test_network.py
ContextSuite context=TestSharedNetwork>:teardown | `Error` | 2.25 | test_network.py
ContextSuite context=TestSharedNetworkWithConfigDrive>:setup | `Error` | 2.40 | test_network.py
test_01_nic | `Error` | 73.20 | test_nic.py
test_01_non_strict_host_anti_affinity | `Error` | 2.92 | test_nonstrict_affinity_group.py
test_02_non_strict_host_affinity | `Error` | 3.65 | test_nonstrict_affinity_group.py
ContextSuite context=TestIsolatedNetworksPasswdServer>:setup | `Error` | 0.00 | test_password_server.py
test_01_isolated_persistent_network | `Error` | 0.27 | test_persistent_network.py
test_02_L2_persistent_network | `Error` | 1.34 | test_persistent_network.py
test_03_deploy_and_destroy_VM_and_verify_network_resources_persist | `Failure` | 2.58 | test_persistent_network.py
test_03_deploy_and_destroy_VM_and_verify_network_resources_persist | `Error` | 2.58 | test_persistent_network.py
ContextSuite context=TestL2PersistentNetworks>:teardown | `Error` | 2.64 | test_persistent_network.py
test_01_create_delete_portforwarding_fornonvpc | `Error` | 4.77 | test_portforwardingrules.py
test_01_add_primary_storage_disabled_host | `Error` | 15.54 | test_primary_storage.py
test_01_primary_storage_iscsi | `Error` | 0.24 | test_primary_storage.py
test_01_primary_storage_nfs | `Error` | 0.22 | test_primary_storage.py
ContextSuite context=TestStorageTags>:setup | `Error` | 0.41 | test_primary_storage.py
test_01_vpc_privategw_acl | `Failure` | 5.63 | test_privategw_acl.py
test_02_vpc_privategw_static_routes | `Failure` | 5.52 | test_privategw_acl.py
test_03_vpc_privategw_restart_vpc_cleanup | `Failure` | 5.34 | test_privategw_acl.py
test_04_rvpc_privategw_static_routes | `Failure` | 5.77 | test_privategw_acl.py
test_02_create_volume | `Error` | 3.26 | test_resource_names.py
ContextSuite context=TestSharedFSLifecycle>:setup | `Error` | 0.00 | test_sharedfs_lifecycle.py
test_01_deploy_vm_on_specific_host | `Error` | 0.10 | test_vm_deployment_planner.py
test_04_deploy_vm_on_host_override_pod_and_cluster | `Error` | 0.12 | test_vm_deployment_planner.py
test_08_migrate_vm | `Error` | 0.07 | test_vm_life_cycle.py
test_01_migrate_vm_strict_tags_success | `Error` | 0.31 | test_vm_strict_host_tags.py
test_02_migrate_vm_strict_tags_failure | `Error` | 0.27 | test_vm_strict_host_tags.py
test_01_restore_vm_strict_tags_success | `Error` | 0.36 | test_vm_strict_host_tags.py
test_02_restore_vm_strict_tags_failure | `Error` | 0.37 | test_vm_strict_host_tags.py
test_01_scale_vm_strict_tags_success | `Error` | 0.30 | test_vm_strict_host_tags.py
test_02_scale_vm_strict_tags_failure | `Error` | 0.33 | test_vm_strict_host_tags.py
test_01_deploy_vm_on_specific_host_without_strict_tags | `Error` | 0.28 | test_vm_strict_host_tags.py
test_02_deploy_vm_on_any_host_without_strict_tags | `Error` | 3.62 | test_vm_strict_host_tags.py
test_03_deploy_vm_on_specific_host_with_strict_tags_success | `Error` | 0.30 | test_vm_strict_host_tags.py
test_04_deploy_vm_on_any_host_with_strict_tags_success | `Error` | 3.72 | test_vm_strict_host_tags.py
test_05_deploy_vm_on_specific_host_with_strict_tags_failure | `Failure` | 0.27 | test_vm_strict_host_tags.py

blueorangutan commented:
<b>[SF] Trillian test result (tid-11330)</b>
Environment: kvm-ol8 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 56929 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11330-kvm-ol8.zip
Smoke tests completed. 138 look OK, 2 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_02_redundant_VPC_default_routes | `Failure` | 422.35 | test_vpc_redundant.py
test_02_VPC_default_routes | `Failure` | 198.78 | test_vpc_router_nics.py

blueorangutan commented:
<b>[SF] Trillian test result (tid-11331)</b>
Environment: kvm-ubuntu22 (x2), Advanced Networking with Mgmt server u22
Total time taken: 58729 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11331-kvm-ubuntu22.zip
Smoke tests completed. 138 look OK, 2 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_03_secured_to_nonsecured_vm_migration | `Error` | 218.55 | test_vm_life_cycle.py
test_03_secured_to_nonsecured_vm_migration | `Error` | 218.55 | test_vm_life_cycle.py
test_04_nonsecured_to_secured_vm_migration | `Error` | 0.02 | test_vm_life_cycle.py
test_08_migrate_vm | `Error` | 53.65 | test_vm_life_cycle.py
test_01_migrate_vm_strict_tags_success | `Error` | 114.65 | test_vm_strict_host_tags.py

rohityadavcloud commented:
@blueorangutan package

blueorangutan commented:
@rohityadavcloud a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✖️ el8 ✖️ el9 ✖️ debian ✖️ suse15. SL-JID 10976

abh1sar commented:
@blueorangutan package

blueorangutan commented:
@abh1sar a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10985

DaanHoogland commented:
@blueorangutan test matrix

blueorangutan commented:
@DaanHoogland a [SL] Trillian-Jenkins matrix job (EL8 mgmt + EL8 KVM, Ubuntu22 mgmt + Ubuntu22 KVM, EL8 mgmt + VMware 7.0u3, EL9 mgmt + XCP-ng 8.2 ) has been kicked to run smoke tests

blueorangutan commented:
<b>[SF] Trillian Build Failed (tid-11358)<b/>

borisstoyanov reviewed (approved):
LGTM, the test failures does not seem related.

DaanHoogland commented:
@blueorangutan package

blueorangutan commented:
@DaanHoogland a [SL] Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result [SF]: ✔️ el8 ✔️ el9 ✔️ debian ✔️ suse15. SL-JID 10994

rohityadavcloud commented:
@blueorangutan test

blueorangutan commented:
@rohityadavcloud a [SL] Trillian-Jenkins test job (ol8 mgmt + kvm-ol8) has been kicked to run smoke tests

rohityadavcloud commented:
Merging this based on previous test runs and code review, will do another round of review after merging this.

Pearl1594 commented on engine/schema/src/main/resources/META-INF/db/schema-41910to42000.sql:
```diff
@@ -158,6 +158,229 @@ WHERE
+    INDEX `i_shared_filesystem__domain_id`(`domain_id`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8;
+
+DROP VIEW IF EXISTS `cloud`.`user_vm_view`;
```
Not sure if we need this here as well as it's already added in the respective view file: https://github.com/apache/cloudstack/pull/9208/files#diff-a5be1b062185fbc803c3f9a03ab8aab590b475a8cafbd46760607a7a071ea606R28

abh1sar replied:
Yes, this is not required. Will remove it in a separate PR.

blueorangutan commented:
<b>[SF] Trillian test result (tid-11372)</b>
Environment: kvm-ubuntu22 (x2), Advanced Networking with Mgmt server u22
Total time taken: 54653 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11372-kvm-ubuntu22.zip
Smoke tests completed. 140 look OK, 0 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---

blueorangutan commented:
<b>[SF] Trillian test result (tid-11373)</b>
Environment: kvm-ol8 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 67723 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11373-kvm-ol8.zip
Smoke tests completed. 131 look OK, 3 have errors, 6 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_mount_shared_fs | `Failure` | 85.08 | test_sharedfs_lifecycle.py
test_resize_shared_fs | `Error` | 0.24 | test_sharedfs_lifecycle.py
test_03_secured_to_nonsecured_vm_migration | `Error` | 404.02 | test_vm_life_cycle.py
test_10_attachAndDetach_iso | `Failure` | 678.04 | test_vm_life_cycle.py
test_02_redundant_VPC_default_routes | `Failure` | 1486.14 | test_vpc_redundant.py
test_03_create_redundant_VPC_1tier_2VMs_2IPs_2PF_ACL_reboot_routers | `Failure` | 474.30 | test_vpc_redundant.py
test_05_rvpc_multi_tiers | `Failure` | 608.73 | test_vpc_redundant.py
test_05_rvpc_multi_tiers | `Error` | 608.75 | test_vpc_redundant.py
all_test_vpc_router_nics | `Skipped` | --- | test_vpc_router_nics.py
all_test_vpc_vpn | `Skipped` | --- | test_vpc_vpn.py
all_test_webhook_delivery | `Skipped` | --- | test_webhook_delivery.py
all_test_webhook_lifecycle | `Skipped` | --- | test_webhook_lifecycle.py
all_test_host_maintenance | `Skipped` | --- | test_host_maintenance.py
all_test_hostha_kvm | `Skipped` | --- | test_hostha_kvm.py

blueorangutan commented:
<b>[SF] Trillian test result (tid-11371)</b>
Environment: xcpng82 (x2), Advanced Networking with Mgmt server a9
Total time taken: 72661 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11371-xcpng82.zip
Smoke tests completed. 134 look OK, 6 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_condensed_drs_algorithm | `Failure` | 175.25 | test_cluster_drs.py
test_02_balanced_drs_algorithm | `Failure` | 184.10 | test_cluster_drs.py
ContextSuite context=TestSharedNetworkWithConfigDrive>:setup | `Error` | 7.49 | test_network.py
test_01_non_strict_host_anti_affinity | `Error` | 213.68 | test_nonstrict_affinity_group.py
test_02_non_strict_host_affinity | `Error` | 107.92 | test_nonstrict_affinity_group.py
test_02_create_volume | `Error` | 2.19 | test_resource_names.py
test_05_scale_vm_dont_allow_disk_offering_change | `Failure` | 65.30 | test_scale_vm.py
test_01_volume_usage | `Error` | 97.91 | test_usage.py

blueorangutan commented:
<b>[SF] Trillian test result (tid-11370)</b>
Environment: vmware-70u3 (x2), Advanced Networking with Mgmt server ol8
Total time taken: 89854 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr9208-t11370-vmware-70u3.zip
Smoke tests completed. 123 look OK, 4 have errors, 13 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_01_deployVMInSharedNetwork | `Failure` | 3609.01 | test_network.py
ContextSuite context=TestSharedNetworkWithConfigDrive>:teardown | `Error` | 3610.25 | test_network.py
test_create_pvlan_network | `Error` | 0.09 | test_pvlan.py
test_02_restore_vm_with_disk_offering | `Error` | 61.17 | test_restore_vm.py
test_03_restore_vm_with_disk_offering_custom_size | `Error` | 54.94 | test_restore_vm.py
ContextSuite context=TestSnapshotRootDisk>:setup | `Error` | 0.00 | test_snapshots.py
ContextSuite context=TestSnapshotStandaloneBackup>:setup | `Error` | 0.00 | test_snapshots.py
all_test_ssvm | `Skipped` | --- | test_ssvm.py
all_test_staticroles | `Skipped` | --- | test_staticroles.py
all_test_storage_policy | `Skipped` | --- | test_storage_policy.py
all_test_templates | `Skipped` | --- | test_templates.py
all_test_update_security_group | `Skipped` | --- | test_update_security_group.py
all_test_usage_events | `Skipped` | --- | test_usage_events.py
all_test_usage | `Skipped` | --- | test_usage.py
all_test_vm_autoscaling | `Skipped` | --- | test_vm_autoscaling.py
all_test_vm_deployment_planner | `Skipped` | --- | test_vm_deployment_planner.py
all_test_vm_life_cycle | `Skipped` | --- | test_vm_life_cycle.py
all_test_vm_lifecycle_unmanage_import | `Skipped` | --- | test_vm_lifecycle_unmanage_import.py
all_test_host_maintenance | `Skipped` | --- | test_host_maintenance.py
all_test_hostha_kvm | `Skipped` | --- | test_hostha_kvm.py
